src/manager.rs

```diff
@@ -0,0 +1,194 @@
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        let mut stmt = try!(connection.prepare("SELECT name, source, is_enabled FROM scripts"));
+        let mut result_map = HashMap::new();
+        try!(stmt.query_map(&[], |row| {
+            let name: String = row.get(0);
+            let source: String = row.get(1);
+            let is_enabled: bool = row.get(2);
+
+            if is_enabled {
+                result_map.insert(
+                    name.clone(),
+                    self.start_script(&name, &source).map(|_| ()));
+            }
+        }));
+        Ok(result_map)
+    }
+
+    /// Attempt to add a new script. The script will be executed and persisted to disk.
+    pub fn put(&mut self, name: &String, source: &String) -> Result<(), Error> {
+        try!(self.start_script(&name, &source));
+
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("INSERT OR REPLACE INTO scripts (name, source, is_enabled)
+                VALUES ($1, $2, $3)", &[name, source, &1])
+            .map(|_| ()).map_err(Error::RusqliteError)
+    }
+
+    /// Enable or disable a script, starting or stopping the script if necessary.
+    pub fn set_enabled(&mut self, name: &String, enabled: bool) -> Result<(), Error> {
+        let source = try!(self.get_source(name));
+        let is_running = self.runners.contains_key(name);
+        if !enabled && is_running {
+            if let Some(mut runner) = self.runners.remove(name) {
+                let (tx, rx) = channel();
+                runner.stop(move |result| {
+                    tx.send(result).unwrap();
+                });
+                let _ = try!(rx.recv());
+            }
+        } else if enabled && !is_running {
+            try!(self.start_script(name, &source));
+        } else {
+            // Already running, or already not running.
+        }
+
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        try!(connection.execute("UPDATE scripts SET is_enabled = $1 WHERE name = $2",
+                                &[&enabled, name]));
+        Ok(())
+    }
+
+    /// Remove a script entirely, stopping it if necessary.
+    pub fn remove(&mut self, name: &String) -> Result<(), Error> {
+        try!(self.set_enabled(name, false));
+        let connection = try!(rusqlite::Connection::open(&self.path));
+        connection.execute("DELETE FROM scripts WHERE name = $1", &[name])
+            .map(|_| ())
+            .map_err(Error::RusqliteError)
+    }
+
+    // Remove all scripts, stopping any running scripts.
+    pub fn remove_all(&mut self) -> Result<(), Error> {
+        // Get a copy of the keys (so that we don't borow `self` twice).
+        let keys: Vec<String> = self.runners.keys().map(|k| k.clone()).collect();
+        // Remove any running scripts.
+        for name in keys {
+            try!(self.remove(&name));
```

Comment:
I believe that we should proceed even if removal of a script fails, and only return an error after having nuked the database. Also, please document that.